[BUG] koordlet UT failed when reporting topology

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!-->

**What happened**:

```log
I1021 02:43:37.222645   51011 states_noderesourcetopology.go:229] start to report node topology
==================
WARNING: DATA RACE
Read at 0x0000050c869a by goroutine 79:
  k8s.io/klog/v2.(*loggingT).header()
      /home/runner/go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:607 +0xc5
  k8s.io/klog/v2.(*loggingT).printf()
      /home/runner/go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:739 +0x84
  k8s.io/klog/v2.Infof()
      /home/runner/go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:1408 +0x34b
  github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer.(*callbackRunner).Start.func1()
      /home/runner/work/koordinator/koordinator/pkg/koordlet/statesinformer/callback_runner.go:138 +0x25b

Previous write at 0x0000050c869a by goroutine 20:
  flag.newBoolValue()
      /opt/hostedtoolcache/go/1.17.13/x64/src/flag/flag.go:[112](https://github.com/koordinator-sh/koordinator/actions/runs/3294352738/jobs/5431778821#step:11:113) +0x2fb
  flag.(*FlagSet).BoolVar()
      /opt/hostedtoolcache/go/1.17.13/x64/src/flag/flag.go:632 +0x307
  k8s.io/klog/v2.InitFlags()
      /home/runner/go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:437 +0x2d6
  k8s.io/component-base/logs.UnsupportedLoggingFlags()
      /home/runner/go/pkg/mod/k8s.io/component-base@v0.22.6/logs/config.go:71 +0x7c
  k8s.io/component-base/logs.BindLoggingFlags()
      /home/runner/go/pkg/mod/k8s.io/component-base@v0.22.6/logs/config.go:56 +0x70
  k8s.io/kubernetes/cmd/kubelet/app/options.AddKubeletConfigFlags()
      /home/runner/go/pkg/mod/k8s.io/kubernetes@v1.22.6/cmd/kubelet/app/options/options.go:547 +0x292b
  github.com/koordinator-sh/koordinator/pkg/util/kubelet.NewKubeletOptions()
      /home/runner/work/koordinator/koordinator/pkg/util/kubelet/kubelet.go:63 +0x215
  github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer.(*nodeTopoInformer).reportNodeTopology()
      /home/runner/work/koordinator/koordinator/pkg/koordlet/statesinformer/states_noderesourcetopology.go:246 +0x554
  github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer.Test_reportNodeTopology()
      /home/runner/work/koordinator/koordinator/pkg/koordlet/statesinformer/states_noderesourcetopology_test.go:319 +0x11d1
  testing.tRunner()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1306 +0x47

Goroutine 79 (running) created at:
  github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer.(*callbackRunner).Start()
      /home/runner/work/koordinator/koordinator/pkg/koordlet/statesinformer/callback_runner.go:127 +0xad
  github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer.(*statesInformer).Run·dwrap·5()
      /home/runner/work/koordinator/koordinator/pkg/koordlet/statesinformer/states_informer.go:136 +0x47

Goroutine 20 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1504 +0x9d1
  main.main()
      _testmain.go:161 +0x324
  runtime.main()
      /opt/hostedtoolcache/go/1.17.13/x64/src/runtime/proc.go:255 +0x226
  github.com/cilium/ebpf/link.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/link/syscalls.go:50 +0x2a8
  github.com/cilium/ebpf/link.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/link/syscalls.go:29 +0x24a
  runtime.doInit()
      /opt/hostedtoolcache/go/1.17.13/x64/src/runtime/proc.go:6498 +0x122
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:426 +0x690
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/prog.go:458 +0x630
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:408 +0x5d0
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:236 +0x570
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:220 +0x510
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:203 +0x4b0
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:185 +0x44e
  runtime.doInit()
      /opt/hostedtoolcache/go/1.17.13/x64/src/runtime/proc.go:6498 +0x122
  github.com/cilium/ebpf/internal/btf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/internal/btf/btf.go:728 +0x1bc
==================
==================
WARNING: DATA RACE
Read at 0x0000050c8698 by goroutine 79:
  k8s.io/klog/v2.(*loggingT).formatHeader()
      /home/runner/go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:627 +0xdd
  k8s.io/klog/v2.(*loggingT).header()
      /home/runner/go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:614 +0x199
  k8s.io/klog/v2.(*loggingT).printf()
      /home/runner/go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:739 +0x84
  k8s.io/klog/v2.Infof()
      /home/runner/go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:1408 +0x34b
  github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer.(*callbackRunner).Start.func1()
      /home/runner/work/koordinator/koordinator/pkg/koordlet/statesinformer/callback_runner.go:138 +0x25b

Previous write at 0x0000050c8698 by goroutine 20:
  flag.newBoolValue()
      /opt/hostedtoolcache/go/1.17.13/x64/src/flag/flag.go:112 +0x35b
  flag.(*FlagSet).BoolVar()
      /opt/hostedtoolcache/go/1.17.13/x64/src/flag/flag.go:632 +0x367
  k8s.io/klog/v2.InitFlags()
      /home/runner/go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:438 +0x338
  k8s.io/component-base/logs.UnsupportedLoggingFlags()
      /home/runner/go/pkg/mod/k8s.io/component-base@v0.22.6/logs/config.go:71 +0x7c
  k8s.io/component-base/logs.BindLoggingFlags()
      /home/runner/go/pkg/mod/k8s.io/component-base@v0.22.6/logs/config.go:56 +0x70
  k8s.io/kubernetes/cmd/kubelet/app/options.AddKubeletConfigFlags()
      /home/runner/go/pkg/mod/k8s.io/kubernetes@v1.22.6/cmd/kubelet/app/options/options.go:547 +0x292b
  github.com/koordinator-sh/koordinator/pkg/util/kubelet.NewKubeletOptions()
      /home/runner/work/koordinator/koordinator/pkg/util/kubelet/kubelet.go:63 +0x215
  github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer.(*nodeTopoInformer).reportNodeTopology()
      /home/runner/work/koordinator/koordinator/pkg/koordlet/statesinformer/states_noderesourcetopology.go:246 +0x554
  github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer.Test_reportNodeTopology()
      /home/runner/work/koordinator/koordinator/pkg/koordlet/statesinformer/states_noderesourcetopology_test.go:319 +0x11d1
  testing.tRunner()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1306 +0x47

Goroutine 79 (running) created at:
  github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer.(*callbackRunner).Start()
      /home/runner/work/koordinator/koordinator/pkg/koordlet/statesinformer/callback_runner.go:127 +0xad
  github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer.(*statesInformer).Run·dwrap·5()
      /home/runner/work/koordinator/koordinator/pkg/koordlet/statesinformer/states_informer.go:136 +0x47

Goroutine 20 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1504 +0x9d1
  main.main()
      _testmain.go:161 +0x324
  runtime.main()
      /opt/hostedtoolcache/go/1.17.13/x64/src/runtime/proc.go:255 +0x226
  github.com/cilium/ebpf/link.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/link/syscalls.go:50 +0x2a8
  github.com/cilium/ebpf/link.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/link/syscalls.go:29 +0x24a
  runtime.doInit()
      /opt/hostedtoolcache/go/1.17.13/x64/src/runtime/proc.go:6498 +0x122
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:426 +0x690
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/prog.go:458 +0x630
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:408 +0x5d0
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:236 +0x570
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:220 +0x510
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:203 +0x4b0
  github.com/cilium/ebpf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/syscalls.go:185 +0x44e
  runtime.doInit()
      /opt/hostedtoolcache/go/1.17.13/x64/src/runtime/proc.go:6498 +0x122
  github.com/cilium/ebpf/internal/btf.init()
      /home/runner/go/pkg/mod/github.com/cilium/ebpf@v0.6.2/internal/btf/btf.go:728 +0x1bc
==================
I1021 02:43:37.225547   51011 callback_runner.go:138] callback runner RegisterTypeNodeSLOSpec loop is exited
I1021 02:43:37.225929   51011 callback_runner.go:138] callback runner RegisterTypeAllPods loop is exited
I1021 02:43:37.225978   51011 callback_runner.go:138] callback runner RegisterTypeNodeTopology loop is exited
--- FAIL: Test_reportNodeTopology (0.00s)
    testing.go:[115](https://github.com/koordinator-sh/koordinator/actions/runs/3294352738/jobs/5431778821#step:11:116)2: race detected during execution of test
```

**What you expected to happen**:

UT passed

**How to reproduce it (as minimally and precisely as possible)**:

Run UT `Test_reportNodeTopology`

**Anything else we need to know?**:

**Environment**:
- App version:
- Kubernetes version (use `kubectl version`):
- Install details (e.g. helm install args):
- Node environment (for koordlet/runtime-proxy issue):
  - Containerd/Docker version:
  - OS version:
  - Kernal version:
  - Cgroup driver: cgroupfs/systemd
- Others:

--- a/pkg/koordlet/statesinformer/kubelet_stub_test.go
+++ b/pkg/koordlet/statesinformer/kubelet_stub_test.go
@@ -18,7 +18,6 @@ package statesinformer
 
 import (
 	"encoding/json"
-	"flag"
 	"log"
 	"net"
 	"net/http"
@@ -80,35 +79,36 @@ func parseHostAndPort(rawURL string) (string, string, error) {
 }
 
 func Test_kubeletStub_GetAllPods(t *testing.T) {
-	flag.StringVar(&token, "token", "mockTest", "")
-	flag.Parse()
+	t.Run("test not panic", func(t *testing.T) {
+		token = "token"
 
-	server := httptest.NewTLSServer(http.HandlerFunc(mockPodsList))
-	defer server.Close()
+		server := httptest.NewTLSServer(http.HandlerFunc(mockPodsList))
+		defer server.Close()
 
-	address, portStr, err := parseHostAndPort(server.URL)
-	if err != nil {
-		t.Fatal(err)
-	}
+		address, portStr, err := parseHostAndPort(server.URL)
+		if err != nil {
+			t.Fatal(err)
+		}
 
-	port, _ := strconv.Atoi(portStr)
-	cfg := &rest.Config{
-		Host:        net.JoinHostPort(address, portStr),
-		BearerToken: token,
-		TLSClientConfig: rest.TLSClientConfig{
-			Insecure: true,
-		},
-	}
+		port, _ := strconv.Atoi(portStr)
+		cfg := &rest.Config{
+			Host:        net.JoinHostPort(address, portStr),
+			BearerToken: token,
+			TLSClientConfig: rest.TLSClientConfig{
+				Insecure: true,
+			},
+		}
 
-	client, err := NewKubeletStub(address, port, "https", 10*time.Second, cfg)
-	if err != nil {
-		t.Fatal(err)
-	}
-	ps, err := client.GetAllPods()
-	if err != nil {
-		t.Fatal(err)
-	}
-	t.Logf("podList %+v\n", ps)
+		client, err := NewKubeletStub(address, port, "https", 10*time.Second, cfg)
+		if err != nil {
+			t.Fatal(err)
+		}
+		ps, err := client.GetAllPods()
+		if err != nil {
+			t.Fatal(err)
+		}
+		t.Logf("podList %+v\n", ps)
+	})
 }
 
 func TestNewKubeletStub(t *testing.T) {

--- a/pkg/koordlet/statesinformer/states_noderesourcetopology.go
+++ b/pkg/koordlet/statesinformer/states_noderesourcetopology.go
@@ -27,12 +27,12 @@ import (
 	"github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
 	topov1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
 	topologyclientset "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/tools/cache"
-	"k8s.io/client-go/util/retry"
 	"k8s.io/klog/v2"
 	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager"
 	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state"
@@ -99,10 +99,6 @@ func (s *nodeTopoInformer) Setup(ctx *pluginOption, state *pluginState) {
 }
 
 func (s *nodeTopoInformer) Start(stopCh <-chan struct{}) {
-	if !features.DefaultKoordletFeatureGate.Enabled(features.NodeTopologyReport) {
-		klog.V(3).Infof("feature %v not enalbed, no need to run module", features.NodeTopologyReport)
-		return
-	}
 	klog.V(2).Infof("starting node topo informer")
 	if !cache.WaitForCacheSync(stopCh, s.nodeInformer.HasSynced, s.podsInformer.HasSynced) {
 		klog.Fatalf("timed out waiting for pod caches to sync")
@@ -119,7 +115,6 @@ func (s *nodeTopoInformer) createNodeTopoIfNotExist() {
 	node := s.nodeInformer.GetNode()
 	topologyName := node.Name
 	ctx := context.TODO()
-	blocker := true
 	_, err := s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Get(ctx, topologyName, metav1.GetOptions{ResourceVersion: "0"})
 	if err == nil {
 		return
@@ -129,33 +124,100 @@ func (s *nodeTopoInformer) createNodeTopoIfNotExist() {
 		return
 	}
 
-	topology := &v1alpha1.NodeResourceTopology{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: topologyName,
-			Labels: map[string]string{
-				extension.LabelManagedBy: "Koordinator",
-			},
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					APIVersion:         "v1",
-					Kind:               "Node",
-					Name:               node.Name,
-					UID:                node.GetUID(),
-					Controller:         &blocker,
-					BlockOwnerDeletion: &blocker,
-				},
-			},
-		},
-		// fields are required
-		TopologyPolicies: []string{string(v1alpha1.None)},
-		Zones:            v1alpha1.ZoneList{v1alpha1.Zone{Name: "fake-name", Type: "fake-type"}},
-	}
+	topology := newNodeTopo(node)
 	// TODO: add retry if create fail
 	_, err = s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Create(ctx, topology, metav1.CreateOptions{})
 	if err != nil {
 		klog.Errorf("failed to create NodeResourceTopology %s, err: %v", topologyName, err)
 		return
 	}
+}
+
+func (s *nodeTopoInformer) calcNodeTopo() (map[string]string, error) {
+	nodeCPUInfo, cpuTopology, sharedPoolCPUs, err := s.calCPUTopology()
+	if err != nil {
+		return nil, fmt.Errorf("failed to calculate cpu topology, err: %v", err)
+	}
+
+	kubeletPort := int(s.nodeInformer.GetNode().Status.DaemonEndpoints.KubeletEndpoint.Port)
+	args, err := getKubeletCommandlineFn(kubeletPort)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Kubelet commandline with kubeletPort %d, err: %v", kubeletPort, err)
+	}
+
+	klog.V(5).Infof("kubelet args: %v", args)
+
+	kubeletOptions, err := kubeletutil.NewKubeletOptions(args)
+	if err != nil {
+		return nil, fmt.Errorf("failed to NewKubeletOptions, err: %v", err)
+	}
+
+	// default policy is none
+	cpuManagerPolicy := extension.KubeletCPUManagerPolicy{
+		Policy:  kubeletOptions.CPUManagerPolicy,
+		Options: kubeletOptions.CPUManagerPolicyOptions,
+	}
+
+	if kubeletOptions.CPUManagerPolicy == string(cpumanager.PolicyStatic) {
+		topology := kubeletutil.NewCPUTopology((*util.LocalCPUInfo)(nodeCPUInfo))
+		reservedCPUs, err := kubeletutil.GetStaticCPUManagerPolicyReservedCPUs(topology, kubeletOptions)
+		if err != nil {
+			klog.Errorf("Failed to GetStaticCPUManagerPolicyReservedCPUs, err: %v", err)
+		}
+		cpuManagerPolicy.ReservedCPUs = reservedCPUs.String()
+
+		// NOTE: We should not remove reservedCPUs from sharedPoolCPUs to
+		//  ensure that Burstable Pods (e.g. Pods request 0C but are limited to 4C)
+		//  at least there are reservedCPUs available when nodes are allocated
+	}
+
+	cpuManagerPolicyJSON, err := json.Marshal(cpuManagerPolicy)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal cpu manager policy, err: %v", err)
+	}
+
+	var podAllocsJSON []byte
+	stateFilePath := kubeletutil.GetCPUManagerStateFilePath(kubeletOptions.RootDirectory)
+	data, err := os.ReadFile(stateFilePath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, fmt.Errorf("failed to read state file, err: %v", err)
+		}
+	}
+	// TODO: report lse/lsr pod from cgroup
+	if len(data) > 0 {
+		podAllocs, err := s.calGuaranteedCpu(sharedPoolCPUs, string(data))
+		if err != nil {
+			return nil, fmt.Errorf("failed to cal GuaranteedCpu, err: %v", err)
+		}
+		if len(podAllocs) != 0 {
+			podAllocsJSON, err = json.Marshal(podAllocs)
+			if err != nil {
+				return nil, fmt.Errorf("failed to marshal pod allocs, err: %v", err)
+			}
+		}
+	}
+
+	cpuTopologyJSON, err := json.Marshal(cpuTopology)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal cpu topology of node, err: %v", err)
+	}
+
+	sharePools := s.calCPUSharePools(sharedPoolCPUs)
+	cpuSharePoolsJSON, err := json.Marshal(sharePools)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal cpushare pools of node, err: %v", err)
+	}
+
+	annotations := map[string]string{}
+	annotations[extension.AnnotationNodeCPUTopology] = string(cpuTopologyJSON)
+	annotations[extension.AnnotationNodeCPUSharedPools] = string(cpuSharePoolsJSON)
+	annotations[extension.AnnotationKubeletCPUManagerPolicy] = string(cpuManagerPolicyJSON)
+	if len(podAllocsJSON) != 0 {
+		annotations[extension.AnnotationNodeCPUAllocs] = string(podAllocsJSON)
+	}
+
+	return annotations, nil
 }
 
 func (s *nodeTopoInformer) calGuaranteedCpu(usedCPUs map[int32]*extension.CPUInfo, stateJSON string) ([]extension.PodCPUAlloc, error) {
@@ -227,113 +289,51 @@ func (s *nodeTopoInformer) calGuaranteedCpu(usedCPUs map[int32]*extension.CPUInf
 
 func (s *nodeTopoInformer) reportNodeTopology() {
 	klog.Info("start to report node topology")
-	s.createNodeTopoIfNotExist()
+	// do not CREATE if reporting is disabled,
+	// but update the node topo object internally
+	if features.DefaultKoordletFeatureGate.Enabled(features.NodeTopologyReport) {
+		s.createNodeTopoIfNotExist()
+	} else {
+		klog.V(5).Infof("feature %v not enabled, node topo will not be reported", features.NodeTopologyReport)
+	}
+
+	nodeTopoAnnotations, err := s.calcNodeTopo()
+	if err != nil {
+		klog.Errorf("failed to calculate node topology, err: %v", err)
+		return
+	}
+
 	ctx := context.TODO()
-	nodeCPUInfo, cpuTopology, sharedPoolCPUs, err := s.calCPUTopology()
-	if err != nil {
-		return
-	}
-
-	kubeletPort := int(s.nodeInformer.GetNode().Status.DaemonEndpoints.KubeletEndpoint.Port)
-	args, err := getKubeletCommandlineFn(kubeletPort)
-	if err != nil {
-		klog.Errorf("Failed to GetKubeletCommandline with kubeletPort %d, err: %v", kubeletPort, err)
-		return
-	}
-
-	klog.V(5).Infof("kubelet args: %v", args)
-
-	kubeletOptions, err := kubeletutil.NewKubeletOptions(args)
-	if err != nil {
-		klog.Errorf("Failed to NewKubeletOptions, err: %v", err)
-		return
-	}
-
-	// default policy is none
-	cpuManagerPolicy := extension.KubeletCPUManagerPolicy{
-		Policy:  kubeletOptions.CPUManagerPolicy,
-		Options: kubeletOptions.CPUManagerPolicyOptions,
-	}
-
-	if kubeletOptions.CPUManagerPolicy == string(cpumanager.PolicyStatic) {
-		topology := kubeletutil.NewCPUTopology((*util.LocalCPUInfo)(nodeCPUInfo))
-		reservedCPUs, err := kubeletutil.GetStaticCPUManagerPolicyReservedCPUs(topology, kubeletOptions)
-		if err != nil {
-			klog.Errorf("Failed to GetStaticCPUManagerPolicyReservedCPUs, err: %v", err)
-		}
-		cpuManagerPolicy.ReservedCPUs = reservedCPUs.String()
-
-		// NOTE: We should not remove reservedCPUs from sharedPoolCPUs to
-		//  ensure that Burstable Pods (e.g. Pods request 0C but are limited to 4C)
-		//  at least there are reservedCPUs available when nodes are allocated
-	}
-
-	cpuManagerPolicyJSON, err := json.Marshal(cpuManagerPolicy)
-	if err != nil {
-		klog.Errorf("failed to marshal cpu manager policy, err: %v", err)
-		return
-	}
-
-	var podAllocsJSON []byte
-	stateFilePath := kubeletutil.GetCPUManagerStateFilePath(kubeletOptions.RootDirectory)
-	data, err := os.ReadFile(stateFilePath)
-	if err != nil {
-		if !os.IsNotExist(err) {
-			klog.Errorf("failed to read file, err: %v", err)
-			return
-		}
-	}
-	// TODO: report lse/lsr pod from cgroup
-	if len(data) > 0 {
-		podAllocs, err := s.calGuaranteedCpu(sharedPoolCPUs, string(data))
-		if err != nil {
-			klog.Errorf("failed to cal GuaranteedCpu, err: %v", err)
-			return
-		}
-		if len(podAllocs) != 0 {
-			podAllocsJSON, err = json.Marshal(podAllocs)
-			if err != nil {
-				klog.Errorf("failed to marshal pod allocs, err: %v", err)
-				return
-			}
-		}
-	}
-
-	cpuTopologyJSON, err := json.Marshal(cpuTopology)
-	if err != nil {
-		klog.Errorf("failed to marshal cpu topology of node, err: %v", err)
-		return
-	}
-
-	sharePools := s.calCPUSharePools(sharedPoolCPUs)
-	cpuSharePoolsJSON, err := json.Marshal(sharePools)
-	if err != nil {
-		klog.Errorf("failed to marshal cpushare pools of node, err: %v", err)
-		return
-	}
-
 	node := s.nodeInformer.GetNode()
-	err = retry.OnError(retry.DefaultBackoff, errors.IsTooManyRequests, func() error {
-		nodeResourceTopology, err := s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Get(ctx, node.Name, metav1.GetOptions{ResourceVersion: "0"})
-		if err != nil {
-			klog.Errorf("failed to get node resource topology %s, err: %v", node.Name, err)
-			return err
+	err = util.RetryOnConflictOrTooManyRequests(func() error {
+		var nodeResourceTopology *v1alpha1.NodeResourceTopology
+		if features.DefaultKoordletFeatureGate.Enabled(features.NodeTopologyReport) {
+			nodeResourceTopology, err = s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Get(ctx, node.Name, metav1.GetOptions{ResourceVersion: "0"})
+			if err != nil {
+				klog.Errorf("failed to get node resource topology %s, err: %v", node.Name, err)
+				return err
+			}
+		} else {
+			nodeResourceTopology = newNodeTopo(node)
 		}
+
+		// set fields
 		if nodeResourceTopology.Annotations == nil {
 			nodeResourceTopology.Annotations = make(map[string]string)
 		}
+		for k, v := range nodeTopoAnnotations {
+			nodeResourceTopology.Annotations[k] = v
+		}
 		// TODO only update if necessary
 		s.updateNodeTopo(nodeResourceTopology)
-		nodeResourceTopology.Annotations[extension.AnnotationNodeCPUTopology] = string(cpuTopologyJSON)
-		nodeResourceTopology.Annotations[extension.AnnotationNodeCPUSharedPools] = string(cpuSharePoolsJSON)
-		nodeResourceTopology.Annotations[extension.AnnotationKubeletCPUManagerPolicy] = string(cpuManagerPolicyJSON)
-		if len(podAllocsJSON) != 0 {
-			nodeResourceTopology.Annotations[extension.AnnotationNodeCPUAllocs] = string(podAllocsJSON)
-		}
-		_, err = s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Update(context.TODO(), nodeResourceTopology, metav1.UpdateOptions{})
-		if err != nil {
-			klog.Errorf("failed to update cpu info of node %s, err: %v", node.Name, err)
-			return err
+
+		// do UPDATE
+		if features.DefaultKoordletFeatureGate.Enabled(features.NodeTopologyReport) {
+			_, err = s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Update(context.TODO(), nodeResourceTopology, metav1.UpdateOptions{})
+			if err != nil {
+				klog.Errorf("failed to update cpu info of node %s, err: %v", node.Name, err)
+				return err
+			}
 		}
 		return nil
 	})
@@ -426,4 +426,29 @@ func (s *nodeTopoInformer) setNodeTopo(newTopo *v1alpha1.NodeResourceTopology) {
 	s.nodeTopoMutex.Lock()
 	defer s.nodeTopoMutex.Unlock()
 	s.nodeTopology = newTopo.DeepCopy()
+}
+
+func newNodeTopo(node *corev1.Node) *v1alpha1.NodeResourceTopology {
+	blocker := true
+	return &v1alpha1.NodeResourceTopology{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: node.Name,
+			Labels: map[string]string{
+				extension.LabelManagedBy: "Koordinator",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         "v1",
+					Kind:               "Node",
+					Name:               node.Name,
+					UID:                node.GetUID(),
+					Controller:         &blocker,
+					BlockOwnerDeletion: &blocker,
+				},
+			},
+		},
+		// fields are required
+		TopologyPolicies: []string{string(v1alpha1.None)},
+		Zones:            v1alpha1.ZoneList{v1alpha1.Zone{Name: "fake-name", Type: "fake-type"}},
+	}
 }

--- a/pkg/koordlet/statesinformer/states_noderesourcetopology_test.go
+++ b/pkg/koordlet/statesinformer/states_noderesourcetopology_test.go
@@ -33,6 +33,7 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager"
 
 	"github.com/koordinator-sh/koordinator/apis/extension"
+	"github.com/koordinator-sh/koordinator/pkg/features"
 	"github.com/koordinator-sh/koordinator/pkg/koordlet/metriccache"
 	mock_metriccache "github.com/koordinator-sh/koordinator/pkg/koordlet/metriccache/mockmetriccache"
 	"github.com/koordinator-sh/koordinator/pkg/util"
@@ -236,100 +237,124 @@ func Test_calGuaranteedCpu(t *testing.T) {
 }
 
 func Test_reportNodeTopology(t *testing.T) {
-	oldFn := getKubeletCommandlineFn
-	defer func() {
-		getKubeletCommandlineFn = oldFn
-	}()
+	t.Run("test not panic", func(t *testing.T) {
+		// prepare feature map
+		enabled := features.DefaultKoordletFeatureGate.Enabled(features.NodeTopologyReport)
+		testFeatureGates := map[string]bool{string(features.NodeTopologyReport): true}
+		err := features.DefaultMutableKoordletFeatureGate.SetFromMap(testFeatureGates)
+		assert.NoError(t, err)
+		defer func() {
+			testFeatureGates[string(features.NodeTopologyReport)] = enabled
+			err = features.DefaultMutableKoordletFeatureGate.SetFromMap(testFeatureGates)
+			assert.NoError(t, err)
+		}()
 
-	client := topologyclientsetfake.NewSimpleClientset()
-	testNode := &corev1.Node{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "test",
-		},
-	}
-	topologyName := testNode.Name
-	mockTopology := v1alpha1.NodeResourceTopology{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: topologyName,
-		},
-		TopologyPolicies: []string{"None"},
-		Zones:            v1alpha1.ZoneList{v1alpha1.Zone{Name: "fake-name", Type: "fake-type"}},
-	}
-	_, err := client.TopologyV1alpha1().NodeResourceTopologies().Create(context.TODO(), &mockTopology, metav1.CreateOptions{})
-	assert.Equal(t, nil, err)
+		oldFn := getKubeletCommandlineFn
+		defer func() {
+			getKubeletCommandlineFn = oldFn
+		}()
 
-	ctl := gomock.NewController(t)
-	mockMetricCache := mock_metriccache.NewMockMetricCache(ctl)
-	mockNodeCPUInfo := metriccache.NodeCPUInfo{
-		ProcessorInfos: []util.ProcessorInfo{
-			{CPUID: 0, CoreID: 0, NodeID: 0, SocketID: 0},
-			{CPUID: 1, CoreID: 0, NodeID: 0, SocketID: 0},
-			{CPUID: 2, CoreID: 1, NodeID: 0, SocketID: 0},
-			{CPUID: 3, CoreID: 1, NodeID: 0, SocketID: 0},
-			{CPUID: 4, CoreID: 2, NodeID: 1, SocketID: 1},
-			{CPUID: 5, CoreID: 2, NodeID: 1, SocketID: 1},
-			{CPUID: 6, CoreID: 3, NodeID: 1, SocketID: 1},
-			{CPUID: 7, CoreID: 3, NodeID: 1, SocketID: 1},
-		},
-	}
+		client := topologyclientsetfake.NewSimpleClientset()
+		testNode := &corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test",
+			},
+		}
+		topologyName := testNode.Name
+		mockTopology := v1alpha1.NodeResourceTopology{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: topologyName,
+			},
+			TopologyPolicies: []string{"None"},
+			Zones:            v1alpha1.ZoneList{v1alpha1.Zone{Name: "fake-name", Type: "fake-type"}},
+		}
+		_, err = client.TopologyV1alpha1().NodeResourceTopologies().Create(context.TODO(), &mockTopology, metav1.CreateOptions{})
+		assert.Equal(t, nil, err)
 
-	mockPodMeta := map[string]*PodMeta{
-		"pod1": {
-			Pod: &corev1.Pod{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "pod1",
-					Namespace: "ns1",
-					Annotations: map[string]string{
-						extension.AnnotationResourceStatus: `{"cpuset": "4-5" }`,
+		ctl := gomock.NewController(t)
+		defer ctl.Finish()
+		mockMetricCache := mock_metriccache.NewMockMetricCache(ctl)
+		mockNodeCPUInfo := metriccache.NodeCPUInfo{
+			ProcessorInfos: []util.ProcessorInfo{
+				{CPUID: 0, CoreID: 0, NodeID: 0, SocketID: 0},
+				{CPUID: 1, CoreID: 0, NodeID: 0, SocketID: 0},
+				{CPUID: 2, CoreID: 1, NodeID: 0, SocketID: 0},
+				{CPUID: 3, CoreID: 1, NodeID: 0, SocketID: 0},
+				{CPUID: 4, CoreID: 2, NodeID: 1, SocketID: 1},
+				{CPUID: 5, CoreID: 2, NodeID: 1, SocketID: 1},
+				{CPUID: 6, CoreID: 3, NodeID: 1, SocketID: 1},
+				{CPUID: 7, CoreID: 3, NodeID: 1, SocketID: 1},
+			},
+		}
+
+		mockPodMeta := map[string]*PodMeta{
+			"pod1": {
+				Pod: &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pod1",
+						Namespace: "ns1",
+						Annotations: map[string]string{
+							extension.AnnotationResourceStatus: `{"cpuset": "4-5" }`,
+						},
 					},
 				},
 			},
-		},
-		"pod2": {
-			Pod: &corev1.Pod{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "pod2",
-					Namespace: "ns2",
-					Annotations: map[string]string{
-						extension.AnnotationResourceStatus: `{"cpuset": "3" }`,
+			"pod2": {
+				Pod: &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pod2",
+						Namespace: "ns2",
+						Annotations: map[string]string{
+							extension.AnnotationResourceStatus: `{"cpuset": "3" }`,
+						},
 					},
 				},
 			},
-		},
-	}
-	mockMetricCache.EXPECT().GetNodeCPUInfo(gomock.Any()).Return(&mockNodeCPUInfo, nil).AnyTimes()
-	r := &nodeTopoInformer{
-		topologyClient: client,
-		metricCache:    mockMetricCache,
-		podsInformer: &podsInformer{
-			podMap: mockPodMeta,
-		},
-		nodeInformer: &nodeInformer{
-			node: testNode,
-		},
-		callbackRunner: NewCallbackRunner(),
-	}
+		}
+		mockMetricCache.EXPECT().GetNodeCPUInfo(gomock.Any()).Return(&mockNodeCPUInfo, nil).Times(1)
+		r := &nodeTopoInformer{
+			topologyClient: client,
+			metricCache:    mockMetricCache,
+			podsInformer: &podsInformer{
+				podMap: mockPodMeta,
+			},
+			nodeInformer: &nodeInformer{
+				node: testNode,
+			},
+			callbackRunner: NewCallbackRunner(),
+		}
 
-	getKubeletCommandlineFn = func(port int) ([]string, error) {
-		tempDir := os.TempDir()
-		args := []string{"/usr/bin/kubelet", fmt.Sprintf("--root-dir=%s", tempDir), "--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf", "--kubeconfig=/etc/kubernetes/kubelet.conf", "--container-log-max-files", "10", "--container-log-max-size=100Mi", "--max-pods", "213", "--pod-max-pids", "16384", "--pod-manifest-path=/etc/kubernetes/manifests", "--network-plugin=cni", "--cni-conf-dir=/etc/cni/net.d", "--cni-bin-dir=/opt/cni/bin", "--v=3", "--enable-controller-attach-detach=true", "--cluster-dns=192.168.0.10", "--pod-infra-container-image=registry-vpc.cn-hangzhou.aliyuncs.com/acs/pause:3.5", "--enable-load-reader", "--cluster-domain=cluster.local", "--cloud-provider=external", "--hostname-override=cn-hangzhou.10.0.4.18", "--provider-id=cn-hangzhou.i-bp1049apy5ggvw0qbuh6", "--authorization-mode=Webhook", "--authentication-token-webhook=true", "--anonymous-auth=false", "--client-ca-file=/etc/kubernetes/pki/ca.crt", "--cgroup-driver=systemd", "--tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_GCM_SHA256", "--tls-cert-file=/var/lib/kubelet/pki/kubelet.crt", "--tls-private-key-file=/var/lib/kubelet/pki/kubelet.key", "--rotate-certificates=true", "--cert-dir=/var/lib/kubelet/pki", "--node-labels=alibabacloud.com/nodepool-id=npab2a7b3f6ce84f5aacc55b08df6b8ecd,ack.aliyun.com=c5558876cbc06429782797388d4abe3e0", "--eviction-hard=imagefs.available<15%,memory.available<300Mi,nodefs.available<10%,nodefs.inodesFree<5%", "--system-reserved=cpu=200m,memory=2732Mi", "--kube-reserved=cpu=1800m,memory=2732Mi", "--kube-reserved=pid=1000", "--system-reserved=pid=1000", "--cpu-manager-policy=static", "--container-runtime=remote", "--container-runtime-endpoint=/var/run/containerd/containerd.sock"}
-		return args, nil
-	}
+		getKubeletCommandlineFn = func(port int) ([]string, error) {
+			tempDir := os.TempDir()
+			args := []string{"/usr/bin/kubelet", fmt.Sprintf("--root-dir=%s", tempDir), "--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf", "--kubeconfig=/etc/kubernetes/kubelet.conf", "--container-log-max-files", "10", "--container-log-max-size=100Mi", "--max-pods", "213", "--pod-max-pids", "16384", "--pod-manifest-path=/etc/kubernetes/manifests", "--network-plugin=cni", "--cni-conf-dir=/etc/cni/net.d", "--cni-bin-dir=/opt/cni/bin", "--v=3", "--enable-controller-attach-detach=true", "--cluster-dns=192.168.0.10", "--pod-infra-container-image=registry-vpc.cn-hangzhou.aliyuncs.com/acs/pause:3.5", "--enable-load-reader", "--cluster-domain=cluster.local", "--cloud-provider=external", "--hostname-override=cn-hangzhou.10.0.4.18", "--provider-id=cn-hangzhou.i-bp1049apy5ggvw0qbuh6", "--authorization-mode=Webhook", "--authentication-token-webhook=true", "--anonymous-auth=false", "--client-ca-file=/etc/kubernetes/pki/ca.crt", "--cgroup-driver=systemd", "--tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_GCM_SHA256", "--tls-cert-file=/var/lib/kubelet/pki/kubelet.crt", "--tls-private-key-file=/var/lib/kubelet/pki/kubelet.key", "--rotate-certificates=true", "--cert-dir=/var/lib/kubelet/pki", "--node-labels=alibabacloud.com/nodepool-id=npab2a7b3f6ce84f5aacc55b08df6b8ecd,ack.aliyun.com=c5558876cbc06429782797388d4abe3e0", "--eviction-hard=imagefs.available<15%,memory.available<300Mi,nodefs.available<10%,nodefs.inodesFree<5%", "--system-reserved=cpu=200m,memory=2732Mi", "--kube-reserved=cpu=1800m,memory=2732Mi", "--kube-reserved=pid=1000", "--system-reserved=pid=1000", "--cpu-manager-policy=static", "--container-runtime=remote", "--container-runtime-endpoint=/var/run/containerd/containerd.sock"}
+			return args, nil
+		}
 
-	r.reportNodeTopology()
+		// reporting enabled
+		r.reportNodeTopology()
 
-	topology, err := client.TopologyV1alpha1().NodeResourceTopologies().Get(context.TODO(), topologyName, metav1.GetOptions{})
-	assert.Equal(t, nil, err)
+		topology, err := client.TopologyV1alpha1().NodeResourceTopologies().Get(context.TODO(), topologyName, metav1.GetOptions{})
+		assert.Equal(t, nil, err)
 
-	expectKubeletCPUManagerPolicy := extension.KubeletCPUManagerPolicy{
-		Policy:       string(cpumanager.PolicyStatic),
-		ReservedCPUs: "0-1",
-	}
-	var kubeletCPUManagerPolicy extension.KubeletCPUManagerPolicy
-	err = json.Unmarshal([]byte(topology.Annotations[extension.AnnotationKubeletCPUManagerPolicy]), &kubeletCPUManagerPolicy)
-	assert.NoError(t, err)
-	assert.Equal(t, expectKubeletCPUManagerPolicy, kubeletCPUManagerPolicy)
+		expectKubeletCPUManagerPolicy := extension.KubeletCPUManagerPolicy{
+			Policy:       string(cpumanager.PolicyStatic),
+			ReservedCPUs: "0-1",
+		}
+		var kubeletCPUManagerPolicy extension.KubeletCPUManagerPolicy
+		err = json.Unmarshal([]byte(topology.Annotations[extension.AnnotationKubeletCPUManagerPolicy]), &kubeletCPUManagerPolicy)
+		assert.NoError(t, err)
+		assert.Equal(t, expectKubeletCPUManagerPolicy, kubeletCPUManagerPolicy)
 
-	assert.Equal(t, `[{"socket":0,"node":0,"cpuset":"0-2"},{"socket":1,"node":1,"cpuset":"6-7"}]`, topology.Annotations[extension.AnnotationNodeCPUSharedPools])
-	assert.Equal(t, `{"detail":[{"id":0,"core":0,"socket":0,"node":0},{"id":1,"core":0,"socket":0,"node":0},{"id":2,"core":1,"socket":0,"node":0},{"id":3,"core":1,"socket":0,"node":0},{"id":4,"core":2,"socket":1,"node":1},{"id":5,"core":2,"socket":1,"node":1},{"id":6,"core":3,"socket":1,"node":1},{"id":7,"core":3,"socket":1,"node":1}]}`, topology.Annotations[extension.AnnotationNodeCPUTopology])
+		assert.Equal(t, `[{"socket":0,"node":0,"cpuset":"0-2"},{"socket":1,"node":1,"cpuset":"6-7"}]`, topology.Annotations[extension.AnnotationNodeCPUSharedPools])
+		assert.Equal(t, `{"detail":[{"id":0,"core":0,"socket":0,"node":0},{"id":1,"core":0,"socket":0,"node":0},{"id":2,"core":1,"socket":0,"node":0},{"id":3,"core":1,"socket":0,"node":0},{"id":4,"core":2,"socket":1,"node":1},{"id":5,"core":2,"socket":1,"node":1},{"id":6,"core":3,"socket":1,"node":1},{"id":7,"core":3,"socket":1,"node":1}]}`, topology.Annotations[extension.AnnotationNodeCPUTopology])
+
+		// reporting disabled
+		testFeatureGates[string(features.NodeTopologyReport)] = false
+		err = features.DefaultMutableKoordletFeatureGate.SetFromMap(testFeatureGates)
+		assert.NoError(t, err)
+		// expect not CREATE/GET/UPDATE any more
+		r.topologyClient = nil
+		mockMetricCache.EXPECT().GetNodeCPUInfo(gomock.Any()).Return(&mockNodeCPUInfo, nil).Times(1)
+		r.reportNodeTopology()
+	})
 }

